packages/react-dom/src/events/SelectEventPlugin.js

```diff
@@ -64,32 +64,44 @@ function getSelection(node) {
       start: node.selectionStart,
       end: node.selectionEnd,
     };
-  } else if (window.getSelection) {
-    var selection = window.getSelection();
-    return {
-      anchorNode: selection.anchorNode,
-      anchorOffset: selection.anchorOffset,
-      focusNode: selection.focusNode,
-      focusOffset: selection.focusOffset,
-    };
+  } else {
+    var win = window;
+    if (node.ownerDocument && node.ownerDocument.defaultView) {
+      win = node.ownerDocument.defaultView;
+    }
+    if (win.getSelection) {
+      var selection = win.getSelection();
+      return {
+        anchorNode: selection.anchorNode,
+        anchorOffset: selection.anchorOffset,
+        focusNode: selection.focusNode,
+        focusOffset: selection.focusOffset,
+      };
+    }
   }
 }
 
 /**
  * Poll selection to see whether it's changed.
  *
  * @param {object} nativeEvent
+ * @param {object} nativeEventTarget
  * @return {?SyntheticEvent}
  */
 function constructSelectEvent(nativeEvent, nativeEventTarget) {
   // Ensure we have the right element, and that the user is not dragging a
   // selection (this matches native `select` event behavior). In HTML5, select
   // fires only on input and textarea thus if there's no focused element we
   // won't dispatch.
+  var doc =
+    nativeEventTarget.ownerDocument ||
+    nativeEventTarget.document ||
+    nativeEventTarget;
```

Comment:
Could you explain what `nativeEventTarget` can be? When would an element be missing an owner document, but have a document? What are the usual values of`nativeEventTarget`?

Reply:
If the event was dispatched at the `window` level, like if you had a `window.onresize` listener, for example, the event target would be the `window` object and would therefore have a `document` but no `ownerDocument`. And if it was dispatched on the `window` object of a nested browsing context, it would be the `window` object of that iframe, not the global one.

However, looking at where this gets invoked, I see that the call site has it’s own nested browsing context-compatible way to calculate `doc` that was added a year ago: https://github.com/facebook/react/commit/6e04bd758ed7a28ebdfbf544f6db9065340392e1

I guess I’ll just repeat the same logic here, if that makes sense to you; i.e. I’ll replace these lines with:

```js
    var doc =
      nativeEventTarget.window === nativeEventTarget
        ? nativeEventTarget.document
        : nativeEventTarget.nodeType === DOCUMENT_NODE
          ? nativeEventTarget
          : nativeEventTarget.ownerDocument;
```

Reply:
Considering they’re in the same file, I’m making a `getEventTargetDocument` function. I dropped `native` from the name because I don’t think that distinction is relevant at the level of this function.